pkgs/development/compilers/gcc/common/builder.nix

```diff
@@ -196,6 +197,13 @@ originalAttrs: (stdenv.mkDerivation (finalAttrs: originalAttrs // {
     mkdir -p "$out/''${targetConfig}/lib"
     mkdir -p "''${!outputLib}/''${targetConfig}/lib"
   '' +
+  # if cross-compiling, link from $lib/lib to $lib/${targetConfig}.
+  # since native-compiles have $lib/lib as a directory (not a
+  # symlink), this ensures that in every case we can assume that
+  # $lib/lib contains the .so files
+  lib.optionalString (with stdenv; targetPlatform.config != hostPlatform.config) ''
+    ln -Ts "''${!outputLib}/''${targetConfig}/lib" $lib/lib
```

Comment:
I'm trying to come up with any adverse side-effects...

I imagine that without this PR we can use some kind of `buildEnv`/`symlinkJoin` to generate FHS prefixes with cross-compilation tools that would somehow approximate what you get with ubuntu-like distributions?..

All in all, I don't see any reason not to merge this, but I lack familiarity with wrapper